internal/translation/datafederation/conversion_endpoints.go

```diff
@@ -6,55 +6,71 @@ import (
 	*akov2.DataFederationPE
 	ProjectID string
 }
 
-func NewDatafederationPrivateEndpointEntry(pe *akov2.DataFederationPE, projectID string) *DatafederationPrivateEndpointEntry {
+func NewPrivateEndpoints(projectID string, privateEndpoints []akov2.DataFederationPE) ([]PrivateEndpoint, error) {
+	entries := make([]PrivateEndpoint, 0, len(privateEndpoints))
+	for _, privateEndpoint := range privateEndpoints {
+		entries = append(entries, NewPrivateEndpoint(projectID, privateEndpoint.DeepCopy()))
+	}
+
+	if err := cmp.Normalize(entries); err != nil {
+		return nil, fmt.Errorf("error normalizing data federation private endpoints: %w", err)
+	}
+
+	return entries, nil
+}
+
+func NewPrivateEndpoint(projectID string, pe *akov2.DataFederationPE) PrivateEndpoint {
 	if pe == nil {
-		return nil
+		return PrivateEndpoint{}
 	}
-	return &DatafederationPrivateEndpointEntry{DataFederationPE: pe, ProjectID: projectID}
+
+	return PrivateEndpoint{DataFederationPE: pe, ProjectID: projectID}
 }
 
-func endpointsFromAtlas(endpoints []admin.PrivateNetworkEndpointIdEntry, projectID string) ([]*DatafederationPrivateEndpointEntry, error) {
-	result := make([]*DatafederationPrivateEndpointEntry, 0, len(endpoints))
+func endpointsFromAtlas(projectID string, endpoints []admin.PrivateNetworkEndpointIdEntry) ([]PrivateEndpoint, error) {
+	entries := make([]PrivateEndpoint, 0, len(endpoints))
 	for _, entry := range endpoints {
-		result = append(result, endpointFromAtlas(&entry, projectID))
+		entries = append(entries, endpointFromAtlas(projectID, &entry))
 	}
-	if err := cmp.Normalize(result); err != nil {
+
+	if err := cmp.Normalize(entries); err != nil {
 		return nil, fmt.Errorf("error normalizing data federation private endpoints: %w", err)
 	}
-	return result, nil
+
+	return entries, nil
 }
 
-func endpointFromAtlas(endpoint *admin.PrivateNetworkEndpointIdEntry, projectID string) *DatafederationPrivateEndpointEntry {
-	result := &DatafederationPrivateEndpointEntry{
-		ProjectID: projectID,
+func endpointFromAtlas(projectID string, endpoint *admin.PrivateNetworkEndpointIdEntry) PrivateEndpoint {
+	if endpoint == nil {
+		return PrivateEndpoint{}
 	}
-	if endpoint != nil {
-		result.DataFederationPE = &akov2.DataFederationPE{
+
+	return PrivateEndpoint{
+		ProjectID: projectID,
+		DataFederationPE: &akov2.DataFederationPE{
 			EndpointID: endpoint.GetEndpointId(),
 			Provider:   endpoint.GetProvider(),
 			Type:       endpoint.GetType(),
-		}
+		},
 	}
-	return result
 }
 
-func endpointToAtlas(ep *DatafederationPrivateEndpointEntry) *admin.PrivateNetworkEndpointIdEntry {
+func endpointToAtlas(ep *PrivateEndpoint) *admin.PrivateNetworkEndpointIdEntry {
 	if ep == nil || ep.DataFederationPE == nil {
 		return nil
 	}
 
 	return &admin.PrivateNetworkEndpointIdEntry{
 		EndpointId: ep.EndpointID,
-		Provider:   pointer.MakePtrOrNil(ep.Provider),
-		Type:       pointer.MakePtrOrNil(ep.Type),
+		Provider:   &ep.Provider,
+		Type:       &ep.Type,
```

Comment:
These are not equivalent, why the change?
Pointer conversions using the pointer package are more explicit and safe usually.

Reply:
replaced back.